### PR TITLE
Bump the version to 1.24.0-preview

### PR DESCRIPTION
**Pull request by NMijat1024**
Because https://github.com/Azure/azure-sdk-for-net/pull/5008 was merged after publishing the 1.23.0-preview, I need to change the version.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by dsgouda** [comment]
@NMijat1024  Please resolve merge conflicts

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj [hunk 1]
```diff
@@ -7,12 +7,11 @@
 		<PackageId>Microsoft.Azure.Management.Sql</PackageId>
 		<Description>Azure SQL Management SDK library</Description>
 		<AssemblyName>Microsoft.Azure.Management.Sql</AssemblyName>
-		<Version>1.23.0-preview</Version>
+		<Version>1.24.0-preview</Version>
 		<PackageTags>Microsoft Azure SQL Management;SQL;SQL Management;</PackageTags>
```
Comment (dsgouda) [inline src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: What's the change being introduced in this package?

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
Per our guidelines, please squash commits into a single commit and update the PR. Looks good for the most part but curious if this is a version bump without any changes

**Comment by NMijat1024** [comment]
Resolved @dsgouda

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj (hunk 1 above)
Comment (NMijat1024) [reply src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: I wrote the reason for this PR on the top of the description

**Comment by dsgouda** [comment]
We should probably be using feature branches to avoid such confusion. I also see that there is [this](https://github.com/Azure/azure-sdk-for-net/pull/5005/) PR open, should it be merged before or after this PR.

**Comment by NMijat1024** [comment]
I guess it is okay to merge this PR first, because nuget package version 1.23.0-preview is already published.

**Review by jaredmoo — APPROVED** [review]

**Review by dsgouda — APPROVED** [review]
LGTM will merge on CIs passing

**Comment by NMijat1024** [comment]
@dsgouda Just wait to squash commits. :)

**Comment by dsgouda** [comment]
@NMijat1024 please resolve conflicts and update the PR

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**